src/controllers/publish.ts

```diff
@@ -295,10 +279,223 @@ export const taskList = async (req: Request, res: Response, next: NextFunction)
+            }
+        }
+
+        const dataPreview = await req.swapi.getMeasurePreview(res.locals.dataset.id);
+        if (req.session.errors) {
+            const errors = req.session.errors;
+            req.session.errors = undefined;
+            req.session.save();
+            res.status(500);
+            res.render('publish/measure-preview', {
+                ...dataPreview,
+                measure,
+                errors
+            });
+        } else {
+            res.render('publish/measure-preview', { ...dataPreview, measure });
+        }
+    } catch (err) {
+        logger.error('Failed to get dimension preview', err);
+        next(new NotFoundException());
+    }
+};
+
+export const measureReview = async (req: Request, res: Response, next: NextFunction) => {
+    try {
+        const dataset = singleLangDataset(res.locals.dataset, req.language);
+        const measure = dataset.measure;
+        if (!measure) {
+            logger.error('Failed to find measure in dataset');
+            next(new NotFoundException());
+            return;
+        }
+        let errors: ViewErrDTO | undefined;
+
+        if (req.method === 'POST') {
+            logger.debug(`User has reviewed measure lookup table.`);
+            switch (req.body.confirm) {
+                case 'true':
+                    res.redirect(req.buildUrl(`/publish/${dataset.id}/tasklist`, req.language));
+                    break;
+                case 'goback':
+                    try {
+                        await req.swapi.resetMeasure(dataset.id);
+                        res.redirect(req.buildUrl(`/publish/${dataset.id}/measure`, req.language));
+                    } catch (err) {
+                        const error = err as ApiException;
+                        logger.error(
+                            `Something went wrong trying to reset the dimension with the following error: ${err}`
+                        );
+                        errors = {
+                            status: error.status || 500,
+                            errors: [
+                                {
+                                    field: '',
+                                    message: {
+                                        key: 'errors.dimension_reset'
+                                    }
+                                }
+                            ],
+                            dataset_id: req.params.datasetId
+                        } as ViewErrDTO;
+                    }
+                    break;
+            }
+            return;
+        }
+
+        const dataPreview = await req.swapi.getMeasurePreview(res.locals.dataset.id);
+        if (errors) {
+            res.status(errors.status || 500);
+            res.render('publish/measure-review', { ...dataPreview, measure, review: true, errors });
+        } else {
+            res.render('publish/measure-review', { ...dataPreview, measure, review: true });
+        }
```

Comment:
The pages (should hopefully) only render errors when set, so I'd just always pass the `errors` prop and save a few lines...

```
if (errors) {
  res.status(errors.status || 500);
}
res.render('publish/measure-review', { ...dataPreview, measure, review: true, errors });
```

Reply:
Fixed